spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -384,6 +385,18 @@ public void testExpireNoLockInStore() throws Exception {
 
 	@Test
 	@RedisAvailable
+	public void testExpireDuringSecondObtain() throws Exception {
```

Comment:
Hi @artembilan ! As I understand correctly this test will pass even without your latest changes in the `RedisLockRegistry` class:  when you do `registry.obtain("foo")` in case the Redis key has expired it will return new `RedisLock` object anyway and `assertNotSame(foo, foo1)` will pass.

Here is another use case: 
  1. Thread A obtains LockFoo
  2. Thread A locks LockFoo
  3. LockFoo expired
  4. Another Thread B successfully obtains LockFoo (will never get into the `if (lockInStore == null || !lock.equals(lockInStore)) {` block)
  5. Thread B tries to lock LockFoo and is blocked because the ReentrantLock is still locked by Thread A

What should be the expected behaviour in your opinion? I think that Thread B should be able to obtain and lock the lock which has already expired in Redis.